test/spec.sh

```diff
@@ -7,18 +7,27 @@ set -o nounset
 set -o pipefail
 set -o errexit
 
-readonly DASH=/bin/dash
-readonly BASH=/bin/bash
-readonly MKSH=/bin/mksh
-readonly ZSH=/usr/bin/zsh  # Ubuntu puts it here
+if hash dash 2>/dev/null; then
```

Comment:
Nitpick: it seems to use both `hash` and `which`.

I think something like:

    readonly DASH=$(which dash || echo /bin/sh)

is cleaner.  Apparently you can also use `hash -t` but I never use that.